coordinator/internal/meshapi/meshapi.go

```diff
@@ -100,3 +102,67 @@ func (i *Server) NewMeshCert(ctx context.Context, _ *meshapi.NewMeshCertRequest)
 
 	return resp, nil
 }
+
+// Recover verifies the peers policy and role and recovers the calling Coordinator.
```

Comment:
It's a bit confusing that there are two subjects here ("peer", "calling coordinator"). Maybe rewrite along the lines of 

> Recover provides key material to authenticated workloads with the Coordinator role.